### ISSUE TITLE
(hash-table-p (make-hash-table :test #'eq :weakness :key)) -> nil

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#946]
Expected would be t.

Looking at the class hierarchy:
```lisp
(clos:class-direct-subclasses (first (clos:class-direct-superclasses (find-class 'core:weak-key-hash-table))))
(#<The BUILT-IN-CLASS HASH-TABLE>
 #<The BUILT-IN-CLASS CORE:WEAK-KEY-HASH-TABLE>)
````

perhaps the definition in https://github.com/clasp-developers/clasp/blob/dev/src/lisp/kernel/cleavir/inline.lisp#L182

should be changed 
from `hash-table-p hash-table`
to `hash-table-p  core:hash-table-base`

**Comment by kpoeck** [comment]
Works fine, will create pr, so that Bike can review.
Did also clean up trivial garbage, will make pr there

**Comment by Bike** [comment]
I mentioned this before, but maybe we should just not advertise that we have weak hash tables yet. They're pretty broken. Hash table iterators don't work either, I think.

**Comment by kpoeck** [comment]
I fixed the hash-table-iterartors, see #950 

Obviously does not fix the other problems that weak hash-table might have on the c++ side.

**Comment by kpoeck** [comment]
fixed
